File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/timecounter/TimeCounterFragment.java
package com.didichuxing.doraemonkit.kit.timecounter;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.View;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.kit.core.BaseFragment;
import com.didichuxing.doraemonkit.kit.core.SettingItem;
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter;
import com.didichuxing.doraemonkit.widget.titlebar.HomeTitleBar;

/**
 * @desc: Activity跳转耗时检测首页
 */

public class TimeCounterFragment extends BaseFragment {

    @Override
    protected int onRequestLayout() {
        return R.layout.dk_fragment_time_counter_index;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
    }


    private void initView() {
        HomeTitleBar title = findViewById(R.id.title_bar);
        title.setListener(new HomeTitleBar.OnTitleBarClickListener() {
            @Override
            public void onRightClick() {
                getActivity().finish();
            }
        });
        RecyclerView mSettingList = findViewById(R.id.setting_list);
        mSettingList.setLayoutManager(new LinearLayoutManager(getContext()));
        SettingItemAdapter settingItemAdapter = new SettingItemAdapter(getContext());
        mSettingList.setAdapter(settingItemAdapter);
        settingItemAdapter.append(new SettingItem(R.string.dk_item_time_counter_switch, TimeCounterManager.get().isRunning()));
        settingItemAdapter.append(new SettingItem(R.string.dk_item_time_goto_list));

        settingItemAdapter.setOnSettingItemSwitchListener(new SettingItemAdapter.OnSettingItemSwitchListener() {
            @Override
            public void onSettingItemSwitch(View view, SettingItem data, boolean on) {
                if (data.desc == R.string.dk_item_time_counter_switch) {
                    if (on) {
                        TimeCounterManager.get().start();
                    } else {
                        TimeCounterManager.get().stop();
                    }
                }
            }
        });
        settingItemAdapter.setOnSettingItemClickListener(new SettingItemAdapter.OnSettingItemClickListener() {
            @Override
            public void onSettingItemClick(View view, SettingItem data) {
                if (data.desc == R.string.dk_item_time_goto_list) {
                    showContent(TimeCounterListFragment.class);
                }
            }
        });
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/timecounter/TimeCounterFragment.kt
package com.didichuxing.doraemonkit.kit.timecounter

import android.view.View
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.core.SettingItem
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter
import com.didichuxing.doraemonkit.kit.performance.AbsPerformanceFragment

/**
 * 启动耗时fragment
 *
 * @author yfengtech
 *
 * 2020-07-01 15:56:18
 */
class TimeCounterFragment : AbsPerformanceFragment() {

    private val kitInfo = TimeCounterKit()

    override val title = kitInfo.name
    override val performanceType = 0

    override fun getSettingItems(list: MutableList<SettingItem>): MutableList<SettingItem>? = mutableListOf(
            SettingItem(R.string.dk_item_time_counter_switch, 0,
                    canCheck = true,
                    isChecked = TimeCounterManager.isRunning),
            SettingItem(R.string.dk_item_time_goto_list)
    )

    override val itemSwitchListener = object : SettingItemAdapter.OnSettingItemSwitchListener {
        override fun onSettingItemSwitch(view: View, data: SettingItem, on: Boolean) {
            if (on) {
                TimeCounterManager.start()
            } else {
                TimeCounterManager.stop()
            }
        }
    }
    override val itemClickListener = object : SettingItemAdapter.OnSettingItemClickListener {
        override fun onSettingItemClick(view: View, data: SettingItem) {
            if (data.desc == R.string.dk_item_time_goto_list) {
                // 点击查看记录
                showContent(TimeCounterListFragment::class.java)
            }
        }
    }
}